Improve literal datatype support

langsamu opened the pull request:
Very small change as a first PR, it's a step towards robust literal datatype support in the Jena Model Step as per #2.

Tested with a few combinations of sqlValue/datatype:
![image](https://user-images.githubusercontent.com/240741/105413004-643dfd80-5c2d-11eb-96ab-4e8c344f150e.png)

which produces acceptable RDF:
```turtle
@prefix xsd:     <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs:    <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex:      <http://example.com/> .
@prefix rdf:     <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .

ex:s  a       <> ;
      ex:bnde "1"^^xsd:decimal ;
      ex:bndo "1"^^xsd:double ;
      ex:bnf  "1"^^xsd:float ;
      ex:bnint "1"^^xsd:int ;
      ex:bninte 1 ;
      ex:bns  "1"^^xsd:short ;
      ex:ide  "1"^^xsd:decimal ;
      ex:ido  "1"^^xsd:double ;
      ex:if   "1"^^xsd:float ;
      ex:iint "1"^^xsd:int ;
      ex:iinte 1 ;
      ex:il   "1"^^xsd:long ;
      ex:is   "1"^^xsd:short ;
      ex:l    "1"^^xsd:long ;
      ex:nde  "1"^^xsd:decimal ;
      ex:ndo  "1.0"^^xsd:double ;
      ex:nf   "1.0"^^xsd:float ;
      ex:nint "1"^^xsd:int ;
      ex:ninte 1 ;
      ex:nl   "1"^^xsd:long ;
      ex:ns   "1"^^xsd:short .
```

langsamu committed:
Remove assumption that literals always have string values

This enables input values other than `String` to be made into RDF literals.
Also fix log message for fallback case in literal converter.
Fixes nationalarchives/kettle-jena-plugins#2

langsamu committed:
Reproduce #2

An integration test the exercises the Jena Model Step with a Long converted to an xsd:int property.
Uses StepMockHelper from https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/test/java/org/pentaho/di/trans/steps/mock/StepMockHelper.java

langsamu commented:
[Adding test to reproduce issue](https://github.com/nationalarchives/kettle-jena-plugins/pull/11/commits/9325a21fed4e30439700994f074eb08c51a61a5f) breaks build due to [inconsistent license](https://travis-ci.com/github/nationalarchives/kettle-jena-plugins/jobs/474444315#L276). Test otherwise passes.
Proper solution would be to add (Apache licensed) [StepMockHelper](https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/test/java/org/pentaho/di/trans/steps/mock/StepMockHelper.java) as a dependency, but I couldn't find a package.

adamretter commented:
Excellent work with the mocking!

Can you please add:

1. Maven Failsafe Plugin - so that the Integration Tests are run on `mvn validate`.
2. An Exclusion for the license plugin for the Apache 2 file from Pentaho.

langsamu committed:
Ignore inconsistent license for step mock helper

Exclude StepMockHelper.java from license consistency check.
The helper is part of the Kettle source code but not packaged.
It's licensed under Apache.

langsamu committed:
Run integration tests using Failsafe

adamretter commented on pom.xml:
```diff
@@ -302,6 +302,18 @@
         </executions>
       </plugin>
+      <plugin>
+        <artifactId>maven-failsafe-plugin</artifactId>
```
Please also add the `<groupId>`

adamretter reviewed (changes requested):
Just one small change to the failsafe plugin declaration

langsamu committed:
Add group Id to integration test plugin

langsamu commented:
Failsafe configured as per [Failsafe Usage](https://maven.apache.org/surefire/maven-failsafe-plugin/usage.html), which says integration test should run in verify phase.

adamretter commented:
@langsamu Can you add the phase to the Travis CI config so that Integration Tests are also run by CI please?

langsamu committed:
Run integration tests as well on Travis

adamretter commented:
Good stuff. Thanks @langsamu
